pulumi/grapl/__main__.py

```diff
@@ -66,6 +77,104 @@ def grapl_core_docker_image_tags(
     )
 
 
+# Dynamic Resource Provider Notes
+########################################################################
+# Dynamic resource providers in Python apparently *must* be defined in
+# the __main__.py file due to problems in serializing the
+# implementations.
+#
+# Additionally, despite the documented suggestion to define resource
+# inputs using Python classes, this does not appear to work (again,
+# due to serialization issues). Instead, we can use plain Python
+# dictionaries. Not ideal, but it has the benefit of actually working.
+class ConsulAclBootstrap(Resource):
+    id: pulumi.Output[str]
+    consul_address: pulumi.Output[str]
+    secret_token: pulumi.Output[str]
+
+    def __init__(
+        self,
+        name: str,
+        consul_address: str,
+        secret_token: Optional[str] = None,
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ):
+
+        super().__init__(
+            ConsulAclBootstrapProvider(),
+            name,
+            {
+                "consul_address": consul_address,
+                "secret_token": secret_token,
+            },
+            opts,
+        )
+
+
+class ConsulAclBootstrapProvider(ResourceProvider):
+    # The consul acl bootstrapping can only be run once, after that it returns a 403.
+    # If this is running on a previously bootstrapped cluster, we require a token be configured and create a fake id
+    def create(self, inputs: Mapping[str, Any]) -> CreateResult:
+        response = requests.put(f"{inputs['consul_address']}/v1/acl/bootstrap")
+        if response.status_code == requests.codes.ok:
+            bootstrap_id = response.json()["AccessorID"]
+            secret_id = response.json()["SecretID"]
+        elif response.status_code == requests.codes.forbidden:
+            # We've already run the bootstrap process so instead let's grab the token from the config
+            if inputs["secret_token"] is not None:
+                secret_id = inputs["secret_token"]
+            else:
+                raise Exception(
+                    "If Consul ACL Bootstrapping is complete you must set consul:token with the token"
+                )
+            # Create a fake static id since we don't have the accessor_id available
+            bootstrap_id = "fake-id-since-previously-bootstrapped"
```

Comment:
How does this "fake" data cascade through the system? Is this going to cause problems?